### PR TITLE
Add example for advanced mode with bumper

### LINKED ISSUE #31430: Missing Documentation on how to use Bumper with Ecovacs
### Feedback

So after Upgrading to 2024.2.1 and seeing that HA now supports self hosted solutions for ecovacs, i set up bumper on my network, connected my robots to bumper, i can use my evovacs app with bumper but i can not connect to bumper using HA Config wizard when using custom server. Both on bumper and HA side, there is absolutely no information what to enter in the fields when using self hosted server. Nethertheless i entered some random username, password and my https://portal-eu.ecouser.net as http domain and mqtt://mybroker:8883 as mqtt server but in the response i only got message: `cannot_connect`

### URL

https://www.home-assistant.io/integrations/ecovacs

### Version

2024.2.1

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by duts-ke**
## Proposed change

Adding documentation to the Ecovacs documentation for use with self-hosted instance of bumper
This follows https://github.com/home-assistant/core/issues/110353

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #31430

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/ecovacs.markdown [hunk 1]
```diff
@@ -31,7 +31,7 @@ The `ecovacs` {% term integration %} is the main integration to integrate [Ecova
 
 Additional note: There are some issues during the password encoding. Using some special characters (e.g., `-`) in your password does not work.
 
-With `advanced_mode` enabled, users can use their self-hosted instance over the cloud servers. Self-hosting comes with some requirements and limitations. More information can be found in the [Bumper's documentation](https://bumper.readthedocs.io).
+With `advanced_mode` enabled, users can use their self-hosted instance over the cloud servers. Self-hosting comes with some requirements and limitations. More information can be found in the [Bumper's documentation](https://bumper.readthedocs.io). See [Advanced Mode](#advanced-mode) for additional details.
```
Comment (edenhaus) [inline source/_integrations/ecovacs.markdown]:
```suggestion
With `advanced_mode` enabled, users can use their self-hosted instance over the cloud servers. Self-hosting comes with some requirements and limitations. See [Advanced Mode](#advanced-mode) for additional details.
```

I would move the "More info" sentence to the Advanced Mode section

File: source/_integrations/ecovacs.markdown [hunk 2]
```diff
@@ -128,3 +128,16 @@ Alternatively, you can use the `ecovacs_error` event to watch for errors. This e
 ```
 
 Finally, if a vacuum becomes unavailable (usually due to being idle and off its charger long enough for it to completely power off,) the vacuum's `status` attribute will change to `offline` until it is turned back on.
+
+## Advanced mode
+
+When connecting with to a self-hosted instance you can connect to the server directly using the following settings:
+- `Username`: you can enter any e-mail address
+- `Password`: you can enter any password (as long as it is 6 characters)
```
Comment (edenhaus) [inline source/_integrations/ecovacs.markdown]: Where it's written that the password needs at least 6 characters?

File: source/_integrations/ecovacs.markdown [hunk 3]
```diff
@@ -128,3 +128,16 @@ Alternatively, you can use the `ecovacs_error` event to watch for errors. This e
 ```
 
 Finally, if a vacuum becomes unavailable (usually due to being idle and off its charger long enough for it to completely power off,) the vacuum's `status` attribute will change to `offline` until it is turned back on.
+
+## Advanced mode
+
+When connecting with to a self-hosted instance you can connect to the server directly using the following settings:
+- `Username`: you can enter any e-mail address
```
Comment (edenhaus) [inline source/_integrations/ecovacs.markdown]:
```suggestion
- `Username`: Enter the e-mail address configured in your instance or any valid one if auth is disabled
```

File: source/_integrations/ecovacs.markdown [hunk 4]
```diff
@@ -128,3 +128,16 @@ Alternatively, you can use the `ecovacs_error` event to watch for errors. This e
...
+- `Password`: you can enter any password (as long as it is 6 characters) 
+- `REST URL`: http://`SELF_HOSTED_INSTANCE`:8007
+- `MQTT URL`: mqtt://`SELF_HOSTED_INSTANCE`:8883
```
Comment (edenhaus) [inline source/_integrations/ecovacs.markdown]: Are you sure that it is not `mqtts://` at least for my instance, I need `mqtts`

**Review by edenhaus — CHANGES_REQUESTED** [review]

File: source/_integrations/ecovacs.markdown (hunk 2 above)
Comment (duts-ke) [reply source/_integrations/ecovacs.markdown]: I've found that information here at the bottom of the page: https://bumper.readthedocs.io/en/latest/Use_With_App/

File: source/_integrations/ecovacs.markdown (hunk 4 above)
Comment (duts-ke) [reply source/_integrations/ecovacs.markdown]: you're right. I was able to configure it with mqtt and didn't look any further, but the connection needs mqtts for communication. When using just mqtt the connection times out.
I'll add it to the docs

File: source/_integrations/ecovacs.markdown (hunk 2 above)
Comment (edenhaus) [reply source/_integrations/ecovacs.markdown]: Could you please verify if it is possible to use a shorter one too?
I think this is only a requirement for the app itself

Comment (duts-ke) [reply source/_integrations/ecovacs.markdown]: Just tested it and it doesn't seem to matter... change phrasing to be inline with username

File: source/_integrations/ecovacs.markdown [hunk 5]
```diff
@@ -128,3 +128,16 @@ Alternatively, you can use the `ecovacs_error` event to watch for errors. This e
 ```
 
 Finally, if a vacuum becomes unavailable (usually due to being idle and off its charger long enough for it to completely power off,) the vacuum's `status` attribute will change to `offline` until it is turned back on.
+
+## Advanced mode
+
+When connecting with to a self-hosted instance you can connect to the server directly using the following settings:
+- `Username`: Enter the e-mail address configured in your instance or any valid one if auth is disabled
+- `Password`: Enter the password configured in your instance or any password if auth is disabled
```
Comment (edenhaus) [inline source/_integrations/ecovacs.markdown]:
```suggestion
- `Password`: Enter the password configured in your instance or any string if auth is disabled
```

File: source/_integrations/ecovacs.markdown [hunk 6]
```diff
@@ -128,3 +128,16 @@ Alternatively, you can use the `ecovacs_error` event to watch for errors. This e
 ```
 
 Finally, if a vacuum becomes unavailable (usually due to being idle and off its charger long enough for it to completely power off,) the vacuum's `status` attribute will change to `offline` until it is turned back on.
+
+## Advanced mode
```
Comment (edenhaus) [inline source/_integrations/ecovacs.markdown]:
```suggestion
## Self hosted configuration
```

File: source/_integrations/ecovacs.markdown [hunk 7]
```diff
@@ -31,7 +31,7 @@ The `ecovacs` {% term integration %} is the main integration to integrate [Ecova
 
 Additional note: There are some issues during the password encoding. Using some special characters (e.g., `-`) in your password does not work.
 
-With `advanced_mode` enabled, users can use their self-hosted instance over the cloud servers. Self-hosting comes with some requirements and limitations. More information can be found in the [Bumper's documentation](https://bumper.readthedocs.io).
+With `advanced_mode` enabled, users can use their self-hosted instance over the cloud servers. Self-hosting comes with some requirements and limitations. See [Advanced Mode](#advanced-mode) for additional details.
```
Comment (edenhaus) [inline source/_integrations/ecovacs.markdown]:
```suggestion
With `advanced_mode` enabled, users can use their self-hosted instance over the cloud servers. Self-hosting comes with some requirements and limitations. See [Self hosted configuration](#self-hosted-configuration) for additional details.
```

**Review by edenhaus — CHANGES_REQUESTED** [review]

File: source/_integrations/ecovacs.markdown [hunk 8]
```diff
@@ -31,7 +31,7 @@ The `ecovacs` {% term integration %} is the main integration to integrate [Ecova
 
 Additional note: There are some issues during the password encoding. Using some special characters (e.g., `-`) in your password does not work.
 
-With `advanced_mode` enabled, users can use their self-hosted instance over the cloud servers. Self-hosting comes with some requirements and limitations. More information can be found in the [Bumper's documentation](https://bumper.readthedocs.io).
+With `advanced_mode` enabled, users can use their self-hosted instance over the cloud servers. Self-hosting comes with some requirements and limitations. See [Self hosted configuration](#self-hosted-configuration) for additional details.
```
Comment (c0ffeeca7) [inline source/_integrations/ecovacs.markdown]:
```suggestion
With `advanced_mode` enabled, users can use their self-hosted instance over the cloud servers. Self-hosting comes with some requirements and limitations. See [Self-hosted configuration](#self-hosted-configuration) for additional details.
```

File: source/_integrations/ecovacs.markdown [hunk 9]
```diff
@@ -128,3 +128,16 @@ Alternatively, you can use the `ecovacs_error` event to watch for errors. This e
 ```
 
 Finally, if a vacuum becomes unavailable (usually due to being idle and off its charger long enough for it to completely power off,) the vacuum's `status` attribute will change to `offline` until it is turned back on.
+
+## Self hosted configuration
```
Comment (c0ffeeca7) [inline source/_integrations/ecovacs.markdown]:
```suggestion
## Self-hosted configuration
```

File: source/_integrations/ecovacs.markdown [hunk 10]
```diff
@@ -128,3 +128,16 @@ Alternatively, you can use the `ecovacs_error` event to watch for errors. This e
...
+
+When connecting with to a self-hosted instance you can connect to the server directly using the following settings:
```
Comment (c0ffeeca7) [inline source/_integrations/ecovacs.markdown]:
```suggestion
When connecting to a self-hosted instance, you can connect to the server directly using the following settings:
```

File: source/_integrations/ecovacs.markdown [hunk 11]
```diff
@@ -128,3 +128,16 @@ Alternatively, you can use the `ecovacs_error` event to watch for errors. This e
 ```
 
 Finally, if a vacuum becomes unavailable (usually due to being idle and off its charger long enough for it to completely power off,) the vacuum's `status` attribute will change to `offline` until it is turned back on.
+
+## Self-hosted configuration
+
+When connecting to a self-hosted instance, you can connect to the server directly using the following settings:
+- `Username`: Enter the e-mail address configured in your instance or any valid one if auth is disabled
```
Comment (c0ffeeca7) [inline source/_integrations/ecovacs.markdown]:
```suggestion
- `Username`: Enter the e-mail address configured in your instance. If authentication is disabled, you can enter any valid e-mail address.
```

File: source/_integrations/ecovacs.markdown [hunk 12]
```diff
@@ -128,3 +128,16 @@ Alternatively, you can use the `ecovacs_error` event to watch for errors. This e
...
+- `Password`: Enter the password configured in your instance or any string if auth is disabled
```
Comment (c0ffeeca7) [inline source/_integrations/ecovacs.markdown]:
```suggestion
- `Password`: Enter the password configured in your instance. If authentication is disabled, you can enter any string (series of characters).
```

**Review by c0ffeeca7 — COMMENTED** [review]
Tiny rephrasing

**Review by c0ffeeca7 — DISMISSED** [review]
LGTM, @github-vdabbeel :+1:

File: source/_integrations/ecovacs.markdown [hunk 13]
```diff
@@ -128,3 +128,16 @@ Alternatively, you can use the `ecovacs_error` event to watch for errors. This e
 ```
 
 Finally, if a vacuum becomes unavailable (usually due to being idle and off its charger long enough for it to completely power off,) the vacuum's `status` attribute will change to `offline` until it is turned back on.
+
+## Self-hosted configuration
+
+When connecting to a self-hosted instance, you can connect to the server directly using the following settings:
```
Comment (edenhaus) [inline source/_integrations/ecovacs.markdown]:
```suggestion
Depending on your setup of the self-hosted instance, you can connect to the server using the following settings:
```

**Review by edenhaus — CHANGES_REQUESTED** [review]

**Review by edenhaus — APPROVED** [review]

**Review by edenhaus — APPROVED** [review]
Thanks @github-vdabbeel :+1:

**Pull request merged**